ingestion/src/metadata/ingestion/api/models.py

```diff
@@ -39,3 +44,4 @@ class Either(BaseModel, Generic[T]):
 
     left: Optional[StackTraceError]
     right: Optional[T]
+    middle: Optional[Nothing]
```

Comment:
Let's remove the `middle` change in the `Either` spec. If the ingestion fails we might be missing some validation in what we're yielding

Reply:
In some scenarios, it only skip to process but not an errors. 
For example, if lineage already added so it can be skip to call api to openmetadata server to reduce numbers of api call. So `middle` can help us in some cases like this, wdyt? @pmbrull

Reply:
but in those cases we are not `yield`ing anything, so should be fine since nothing gets to the runner right

Reply:
Removed `Middle` and `Nothing`